test/test_sparse.py

```diff
@@ -246,6 +246,17 @@ def test_sparse_sum():
         ref = test_sparse_sum()
         self.assertTrue(ref.expired())
 
+    @onlyCPU
+    @dtypes(torch.double)
+    def test_coalesce_large_indices(self, device, dtype):
+        # gh-57416
+        N = 100000
+        indices = torch.tensor([[N, N - 1]] * 4, dtype=torch.int64, device=device)
+        values = torch.tensor([1, 2], dtype=dtype, device=device)
+        self.assertRaises(RuntimeError,
+                          lambda: torch.sparse_coo_tensor(indices, values,
+                                                          (N + 1,) * 4, device=device))
```

Comment:
Only the first exception is generated  by `compute_numel` function in this call stack, 
`sparse_coo_tensor/_sparse_coo_tensor_with_dims_and_tensors/resize_/compute_numel`.

Should it be a test also for the other exception, when we want to generate a coalesced sparse tensor?
In this case the call stack is  `sparse_coo_tensor...` and then `_coalesce_sparse_`device`/flatten_indices`

Reply:
yes, the current test catches runtime error only from `comput_numel`.

Since `numel` catches the `product(size)` overflow condition, `coalesce` nor `flatten_indices` are never reached, IIUC.

I kept the overflow checking in `flatten_indices` for the case if `numel` will be disabled/redefined for sparse tensors per https://github.com/pytorch/pytorch/issues/57542 discussion.

Reply:
After a second thought, I re-added the coalesce method call to the test to be sure that the overflow condition will be checked, with or without the numel interference.